src/Orleans.Core/Timers/TimerManager.cs

```diff
@@ -0,0 +1,354 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.Threading;
+
+namespace Orleans.Timers
+{
+    public static class TimerManager
+    {
+        public static Task Delay(TimeSpan timeout) => DelayUntil(DateTime.UtcNow + timeout);
```

Comment:
If we do make an interface for this, suggest Delay be an extension function.

Reply:
makes sense